### PR TITLE
Export multiple versions of the RPC spec

### PR DESCRIPTION
**Pull request by omerfirmak**
Depends on #1242

### CODE REVIEW

**Comment by joshklop** [comment]
Made a quick read and it looks good to me. I'll review once #1242 is merged so the diff is easier to read.

**Comment by omerfirmak** [comment]
Rebased

File: node/http.go [hunk 1]
```diff
@@ -61,25 +60,41 @@ func makeHTTPService(host string, port uint16, handler http.Handler) *httpServic
...
 	if metricsEnabled {
-		httpHandler.WithListener(makeHTTPMetrics())
+		listener = makeHTTPMetrics()
 	}
+
 	mux := http.NewServeMux()
-	mux.Handle("/", httpHandler)
-	mux.Handle("/v0_4", httpHandler)
+	for path := range servers {
+		httpHandler := jsonrpc.NewHTTP(servers[path], log)
```
Comment (joshklop) [inline node/http.go:73]:
```suggestion
	for path, server := range servers {
		httpHandler := jsonrpc.NewHTTP(server, log)
```

File: rpc/handlers.go [hunk 2]
```diff
@@ -1317,6 +1317,10 @@ func (h *Handler) callAndLogErr(f func() error, msg string) {
 	}
 }
 
+func (h *Handler) SpecVersion() (string, *jsonrpc.Error) {
+	return "v0.5.0-rc0", nil
```
Comment (joshklop) [inline rpc/handlers.go:1321]: Don't we implement `v0.4`? I might have lost track of the recent changes.

**Review by joshklop — APPROVED** [review]

**Pull request merged**